pl_bolts/models/self_supervised/simclr/simclr_finetuner.py

```diff
@@ -3,54 +3,162 @@
 
-from pl_bolts.models.self_supervised import SimCLR, SSLFineTuner
-from pl_bolts.models.self_supervised.simclr.transforms import SimCLREvalDataTransform, SimCLRTrainDataTransform
+from pl_bolts.models.self_supervised.ssl_finetuner import SSLFineTuner
+from pl_bolts.models.self_supervised.simclr.simclr_module import SimCLR
+from pl_bolts.models.self_supervised.simclr.transforms import SimCLRFinetuneTransform
+from pl_bolts.transforms.dataset_normalizations import (
+    imagenet_normalization,
+    stl10_normalization,
+    cifar10_normalization
+)
 
 
 def cli_main():  # pragma: no-cover
-    from pl_bolts.datamodules import CIFAR10DataModule, ImagenetDataModule, STL10DataModule
+    from pl_bolts.datamodules import ImagenetDataModule, STL10DataModule, CIFAR10DataModule
 
     pl.seed_everything(1234)
 
     parser = ArgumentParser()
-    parser = pl.Trainer.add_argparse_args(parser)
-    parser.add_argument('--dataset', type=str, help='stl10, cifar10', default='cifar10')
+    parser.add_argument('--dataset', type=str, help='cifar10, stl10, imagenet', default='cifar10')
     parser.add_argument('--ckpt_path', type=str, help='path to ckpt')
-    parser.add_argument('--data_dir', type=str, help='path to ckpt', default=os.getcwd())
-    args = parser.parse_args()
+    parser.add_argument('--data_path', type=str, help='path to ckpt', default=os.getcwd())
+
+    parser.add_argument("--batch_size", default=64, type=int, help="batch size per gpu")
+    parser.add_argument("--num_workers", default=8, type=int, help="num of workers per GPU")
+    parser.add_argument("--gpus", default=4, type=int, help="number of GPUs")
+    parser.add_argument('--num_epochs', default=100, type=int, help="number of epochs")
+
+    # fine-tuner params
+    parser.add_argument('--in_features', type=int, default=2048)
+    parser.add_argument('--dropout', type=float, default=0.)
+    parser.add_argument('--learning_rate', type=float, default=0.3)
+    parser.add_argument('--weight_decay', type=float, default=1e-6)
+    parser.add_argument('--nesterov', type=bool, default=False)
+    parser.add_argument('--scheduler_type', type=str, default='cosine')
+    parser.add_argument('--gamma', type=float, default=0.1)
+    parser.add_argument('--final_lr', type=float, default=0.)
 
-    # load the backbone
-    backbone = SimCLR.load_from_checkpoint(args.ckpt_path, strict=False)
+    args = parser.parse_args()
 
-    # init default datamodule
     if args.dataset == 'cifar10':
-        dm = CIFAR10DataModule.from_argparse_args(args)
-        dm.train_transforms = SimCLRTrainDataTransform(32)
-        dm.val_transforms = SimCLREvalDataTransform(32)
-        dm.test_transforms = SimCLREvalDataTransform(32)
-        args.num_samples = dm.num_samples
+        dm = CIFAR10DataModule(
+            data_dir=args.data_path,
+            batch_size=args.batch_size,
+            num_workers=args.num_workers
+        )
+
+        dm.train_transforms = SimCLRFinetuneTransform(
```

Comment:
well in fact this shall be in the dm as `def set_finetune_transfroms(self) -> None`

Reply:
maybe @akihironitta could take it in another PR...